tests/cron.go

```diff
@@ -475,7 +475,7 @@ func (s *ClientFunctionalSuite) TestCronWorkflowCompletionStates() {
 	ts = time.Now()
 
 	// let first run finish, then check execution and history of second run
-	time.Sleep(500 * time.Millisecond)
+	time.Sleep(2500 * time.Millisecond)
 	exec = s.listOpenWorkflowExecutions(startTs, time.Now(), id, 1)[0]
 	s.Equal(enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING, exec.GetStatus())
```

Comment:
Rather than waiting 2.5s we should poll this periodically. You can use `s. Eventually(condition func() bool, waitFor time.Duration, tick time.Duration, msgAndArgs ...interface{})` to do this

Reply:
So, the thing is I don't know what exact is this waiting for. But I agree that using `s.Eventually` is certainly better.
cc: @dnr

Reply:
I'd wager you're safe to use the next two lines as the Eventually predicate, like

```go
s.Eventually(func() bool {
    exec = s.listOpenWorkflowExecutions(startTs, time.Now(), id, 1)[0]
    s.Equal(enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING, exec.GetStatus())
    return true
}, 5 * time.Second, 100 * time.Millisecond, "Workflow never began executing")
```

Reply:
This won't work. I've updated the PR with the right condition, I think.